### DISCUSSION TITLE
articles/programming-is-still-hard-after-14-years/

### DISCUSSION THREAD

**Discussion by giscus** [piglei/piglei-giscus-comments#5] [General]
# articles/programming-is-still-hard-after-14-years/

piglei 的 blog

https://www.piglei.com/articles/programming-is-still-hard-after-14-years/

**Comment by Yeijon** [comment]
很有启发，感谢作者的分享！

**Comment by wangshangxianwangshangxian** [comment]
很有感触，每次新进入一家公司，看完项目的代码后，就会不自觉产生抵触情绪。

抵触的不是产品经理，也不是时常改来改去的需求，而是那一堆高度耦合又不可拓展的代码。

每次遇到实在难改的需求时，就会安慰自己的，“不管怎么样事情总会被解决的，这种难受的感觉最多只会持续到迭代那天”，我也想写易读的代码，但我又不是这个项目的创始人～堆成山的代码我敢动嘛？

**Comment by x1uc** [comment]
Good Essay, 大学已经快毕业了，单元测试真实没怎么看到过，包括现在实习的过程中也没有看到过，只在一些tutorial中看到过相关的字眼，看到博主对单测有这么高的评价（如果以“开始写单元测试”作为分界线，把我的职业生涯分割成两段，后面那段远比前面那段精彩得多），现在就去了解一下。

**Comment by sweet2honey** [comment]
谢谢分享

**Comment by VictoryLiang** [comment]
感谢分享
